internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -908,6 +910,87 @@ func (b *BeatsMonitor) injectMetricsInput(cfg map[string]interface{}, componentI
 		},
 	}
 
+	// add system/process metrics for services that can't be monitored via json/beats metrics
+	// If there's a checkin PID and it contains the "endpoint" string, assume we want to monitor it
```

Comment:
Instead of just "endpoint" can you look for inputs using the service runtime https://github.com/elastic/elastic-agent/blob/e401afe9d7b3ac9f5b2aef7999642f8194d9d243/specs/endpoint-security.spec.yml#L23

That will remove the hard coding of the input name here and generalize better.

Reply:
So, just to clarify, it looks like there's a component to the input spec: 
https://github.com/elastic/elastic-agent/blob/6573c38318ced4906c2bab2e43c1f72a00d3f210/pkg/component/input_spec.go#L25

That's only populated by endpoint and other components that we'd need to separately catch the PID for?

Reply:
Yes that means it is going to be installed as an independent service and not a sub-process.